### PR TITLE
Move WINVER definition to cmake and a bit of cleanup

### PR DESCRIPTION
**Pull request by freiro**
After a discussion with @JayFoxRox, we agreed to move _WINVER_ definition to CMakeFiles.txt to ensure correct propagation.
To avoid dualities I cleaned up the code a bit.

P.S. Should we make -DNOMINMAX available also to MINGW?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -182,15 +182,15 @@ IF (APPLE)
...
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
-ELSEIF(MINGW)
-    # PSAPI is the Process Status API
-    set(PLATFORM_LIBRARIES winmm ws2_32 psapi imm32 version)
-
-    # WSAPoll functionality doesn't exist before WinNT 6.x (Vista and up)
-    add_definitions(-D_WIN32_WINNT=0x0600)
-ELSEIF(WIN32)
+ELSEIF (WIN32)
+    #WSAPoll and other fileutil.cpp functionalities didn't exist before WinNT 6.x (Vista and up)
```
Comment (JayFoxRox) [inline CMakeLists.txt]: This also affects AppData stuff now?

File: CMakeLists.txt [hunk 2]
```diff
@@ -182,15 +182,15 @@ IF (APPLE)
...
+    add_definitions(-D_WIN32_WINNT=0x0600 -DWINVER=0x0600)
     set(PLATFORM_LIBRARIES winmm ws2_32)
-ELSEIF(CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU|SunOS)$")
+    if (MINGW)
+        # PSAPI is the Process Status API
+        set(PLATFORM_LIBRARIES winmm ws2_32 psapi imm32 version)
+    ENDIF (MINGW)
+ELSEIF (CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU|SunOS)$")
     set(PLATFORM_LIBRARIES rt)
 ENDIF (APPLE)
```
Comment (JayFoxRox) [inline CMakeLists.txt]: unrelated to PR: Why does this say `ENDIF (APPLE)`? Kind of misleading..

File: CMakeLists.txt (hunk 1 above)
Comment (freiro) [inline CMakeLists.txt]: This has always affected AppData stuff, before Windows Vista the path name was completely different and required different functions to retrieve them.
Anyway according to the previous cmakefile we were already targeting Windows Vista or later.

**Comment by NicolaiVdS** [comment]
Can confirm build works on MacOs 10.12

File: CMakeLists.txt (hunk 2 above)
Comment (TTFH) [inline CMakeLists.txt]: Thats from January 2015, before it was:
`IF (APPLE)
    # CoreFoundation is required only on OSX
    FIND_LIBRARY(COREFOUNDATION_LIBRARY CoreFoundation)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
ENDIF (APPLE)`

File: CMakeLists.txt [hunk 3]
```diff
@@ -182,15 +182,15 @@ IF (APPLE)
...
-    add_definitions(-D_WIN32_WINNT=0x0600)
-ELSEIF(WIN32)
+ELSEIF (WIN32)
+    #WSAPoll and other fileutil.cpp functionalities didn't exist before WinNT 6.x (Vista and up)
+    add_definitions(-D_WIN32_WINNT=0x0600 -DWINVER=0x0600)
     set(PLATFORM_LIBRARIES winmm ws2_32)
-ELSEIF(CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU|SunOS)$")
+    if (MINGW)
+        # PSAPI is the Process Status API
+        set(PLATFORM_LIBRARIES winmm ws2_32 psapi imm32 version)
```
Comment (lioncash) [inline CMakeLists.txt]: He meant doing the following:

```cmake
set(PLATFORM_LIBRARIES winmm ws2_32)
if (MINGW)
    # PSAPI is the Process Status API
    list(APPEND PLATFORM_LIBRARIES psapi imm32 version)
```

**Comment by yuriks** [comment]
LGTM

**Pull request merged**